ql/src/java/org/apache/hadoop/hive/ql/exec/Registry.java

```diff
@@ -583,7 +583,13 @@ public void unregisterFunction(String functionName) throws HiveException {
         }
         mFunctions.remove(functionName);
         fi.discarded();
+        FunctionResource[] resources = fi.getResources();
         if (fi.isPersistent()) {
+          Map<String, String> udfCacheMap = SessionState.getUDFCacheMap();
+          for(FunctionResource fr : resources){
+            //remove from udf cache if it's saved.
+            udfCacheMap.remove(fr.getResourceURI());
```

Comment:
Do we need to remove the downloaded file here too?

Reply:
Yeah we need to clear the downloaded files, when unregistering a function.